src/drivers/amqp/Queue.php

```diff
@@ -63,8 +63,13 @@ public function listen()
         $this->open();
         $callback = function(AMQPMessage $payload) {
             list($ttr, $message) = explode(';', $payload->body, 2);
-            if ($this->handleMessage(null, $message, $ttr, 1)) {
-                $payload->delivery_info['channel']->basic_ack($payload->delivery_info['delivery_tag']);
+            $isSuccess = $this->handleMessage(null, $message, $ttr, 1);
+            /** @var AMQPChannel $channel */
+            $channel = $payload->delivery_info['channel'];
+            if ($isSuccess) {
+                $channel->basic_ack($payload->delivery_info['delivery_tag']);
+            } else {
+                $channel->basic_reject($payload->delivery_info['delivery_tag'], true);
```

Comment:
This isn't very good to simply do re-queue. If you have a single consumer the message is immediately re-delivered to the same consumer and you won't get a chance to process any others. The queue is stacked. 

#158 reject the original message and publish the copy with a delay. (it gets to the end of the queue)

Reply:
Yes, you right. I hope your PR will be merged soon.